Makefile

```diff
@@ -1,137 +1,205 @@
-PHP_CS_FIXER_FUTURE_MODE=1
-PHPSTAN=./phpstan.phar
-PHP-CS-FIXER=./php-cs-fixer-v2.phar
-PHPUNIT=vendor/bin/phpunit
-INFECTION=build/bin/infection.phar
+.DEFAULT_GOAL := help
+
+.PHONY: help
+# TODO: add a check to ensure the result is always the expected one
+# TODO: add a check to ensure whenever a PHONY is declared, the following target has the same name; e.g. here .PHONY=help
+# so the next target should be `help:`. Otherwise this is most certainly a mistake.
+help:
+	@echo "\033[33mUsage:\033[0m\n  make TARGET\n\n\033[32m#\n# Commands\n#---------------------------------------------------------------------------\033[0m\n"
+	@fgrep -h "##" $(MAKEFILE_LIST) | fgrep -v fgrep | sed -e 's/\\$$//' | sed -e 's/##//' | awk 'BEGIN {FS = ":"}; {printf "\033[33m%s:\033[0m%s\n", $$1, $$2}'
+
 
-# URLs to download all tools
-BOX_URL="https://github.com/humbug/box/releases/download/3.1.0/box.phar"
-PHP-CS-FIXER_URL="https://cs.sensiolabs.org/download/php-cs-fixer-v2.phar"
+#
+# Variables
+#---------------------------------------------------------------------------
+BOX=./.tools/box
+BOX_URL="https://github.com/humbug/box/releases/download/3.8.0/box.phar"
```

Comment:
upgrade Box to 3.8 on the way... It was a really minor change so I don't think it pollutes that PR too much :)